File: src/leetcode-1/code.ts
export function twoSum(nums: number[], target: number): number[] | undefined {
	const memory = new Map<number, number>();
	for (const [index, num] of nums.entries()) {
		const numIndex = memory.get(num);
		if (numIndex !== undefined) return [numIndex, index];
		memory.set(target - num, index);
	}
}

File: src/leetcode-1/go/code.go
package main

func TwoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, num := range nums {
		val, ok := m[num]
		if ok {
			return []int{val, i}
		}
		complement := target - num
		m[complement] = i
	}
	return nil
}
